File: Design Patterns/Chain of Responsibility Pattern/SeniorDoctor.java
package seniordoctor;

import patient.Patient;
import diseaseseverity.DiseaseSeverity;
import handler.Handler;

public class SeniorDoctor extends Handler{
	
	@Override
	public void handlePatient(Patient patient){
		
		if(patient.getDiseaseSeverity() == DiseaseSeverity.HIGH){
			System.out.println("Diagnose the Patient " + patient.getName() + " by the Senior Doctor \n");
		}else{
			System.out.println("Passing the Patient " + patient.getName() + " to Higher Authority from Senior Doctor");
			this.successor.handlePatient(patient);
		}
		
	}
	
}

File: Design Patterns/Chain of Responsibility Pattern in JavaScript/SeniorDoctor.js
const Handler = require('./Handler');
const DiseaseSeverity = require('./DiseaseSeverity');

class SeniorDoctor extends Handler {

    handlePatient(patient) {

        super.handlePatient(patient);

        if (patient.diseaseSeverity === DiseaseSeverity.HIGH) {
            console.log("Diagnose the Patient " + patient.name + " by the Senior Doctor \n");
        } else {
            console.log("Passing the Patient " + patient.name + " to Higher Authority from Senior Doctor");
            this.successor.handlePatient(patient);
        }

    }

}

module.exports = SeniorDoctor;